src/src/WebServer/ESPEasy_WebServer.cpp

```diff
@@ -407,11 +415,19 @@ void getWebPageTemplateDefault(const String& tmplName, WebTemplateParser& parser
   else if (tmplName.equals(F("TmplDsh")))
   {
     getWebPageTemplateDefaultHead(parser, !addMeta, addJS);
-    parser.process(F(
-      "<body>"
-      "{{content}}"
-      "</body></html>"
-      ));
+    String body;
+    body += F("<body");
+    #if FEATURE_AUTO_DARK_MODE
+    if (0 == Settings.getCssMode()) {
+      body += F(" data-theme='auto'");
+    } else if (2 == Settings.getCssMode()) {
+      body += F(" data-theme='dark'");
+    }
+    #endif // FEATURE_AUTO_DARK_MODE
+    body += F(">"
+              "{{content}}"
+              "</body></html>");
+    parser.process(body);
```

Comment:
Why allocate a String?
You can parse the flash strings directly.

Reply:
To accommodate the `if`'s enclosed in the `#if` above that line. I've been thinking about refactoring the construction of that `"<body..."` part into a separate function, as it is used a few times there.

Reply:
Fixed